tcms/testruns/views.py

```diff
@@ -271,36 +272,71 @@ def _walk_executions(test_executions):
                comments_subtotal.get(case_run.pk, 0))
 
 
-@permission_required('testruns.change_testrun')
-def edit(request, run_id):
+@method_decorator(permission_required('testruns.change_testrun'), name='dispatch')
+class TestRunEditView(FormView):
     """Edit test plan view"""
 
-    try:
-        test_run = TestRun.objects.select_related().get(run_id=run_id)
-    except ObjectDoesNotExist:
-        raise Http404
+    template_name = 'testruns/mutable.html'
+    form_class = BaseRunForm
+
+    def form_invalid(self, form):
+        test_run = self._get_test_run_or_fail(self.kwargs['run_id'])
+
+        context_data = {
+            'test_run': test_run,
+            'test_plan': test_run.plan,
+            'form': self._get_base_form(test_run),
+            'errors': form.errors
+        }
 
-    # If the form is submitted
-    if request.method == "POST":
-        form = BaseRunForm(request.POST)
+        return render(self.request, self.template_name, context_data)
+
+    def form_valid(self, form):
+        run_id = self.kwargs['run_id']
+        test_run = self._get_test_run_or_fail(run_id)
+        test_run.summary = form.cleaned_data['summary']
+        # Permission hack
+        if test_run.manager == self.request.user or test_run.plan.author == self.request.user:
+            test_run.manager = form.cleaned_data['manager']
+        test_run.default_tester = form.cleaned_data['default_tester']
+        test_run.build = form.cleaned_data['build']
+        test_run.product_version = test_run.plan.product_version
+        test_run.notes = form.cleaned_data['notes']
```

Comment:
I'm wondering if there isn't a ready solution to this. Seems counter intuitive to have to manually assign values to each field. At the very worst we can iterate over the form attributes and use `setattrib` on the object, but there must be something to do this for us.

Reply:
I have to dig deeper into this.

Reply:
FYI UpdateView will make this go away, you can either give it a model (and it will build the form dynamically, so no need for a form class) or you give it a list of fields which are allowed to be modified.